docs/concepts.md

```diff
@@ -0,0 +1,85 @@
+
+```cpp
+ddc::DiscreteElement<DDimY> y(0);
+```
+
+carries the strong typing of the ordinate dimension `DDimY` and corresponds to y<sub>0</sub> , the first point along the `Y` dimension.
+
+Moreover, `ddc::DiscreteElement` are very useful for another reason. If we take the example of a classic container in C++, let's say we want to access the element `(i,j)` of this container which corresponds to the grid point at the i th row and j th column, we would do it like this:
+
+```cpp
+container(i,j);
+```
+
+Now, if we take a slice of this container and still want to access the same element `(i,j)` from the grid, we will need to adjust the indices because the indexing of the new sliced container along each dimension starts at zero. However, with ddc, this is not the case. If we take a slice of a chunkspan, accessing a `ddc::DiscreteElement` is the same between the slice and the original chunkspan because of the uniqueness of each discrete element on the grid and because of the way we access data using ddc.
+
+### ddc::Coordinate
+
+A `ddc::Coordinate` is one of the only ddc types which represents a continuous data type. This means that it can take any value that can be represented by a double. It represents the position of a coordinate in the vector space.
+
+Let's take the same discretization of the Y-axis {y<sub>0</sub>, y<sub>1</sub>, ..., y<sub>n</sub>}. We saw that ```cpp ddc::DiscreteElement<DDimY> y(0)``` points to the first element y<sub>0</sub> along the Y-axis. Now if we take ```cpp ddc::Coordinate<DDimY> (ddc::DiscreteElement<DDimY> y(0))``` it will correspond to the actual value of y<sub>0</sub> in the simulation. If the first point is at a position `y = -1`, it will be characterized by a `ddc::DiscreteElement` as the first point of the discretization, but physically its position is -1 and this is the value that `ddc::Coordinate` of this element will be.
+
+The `ddc::Coordinate` thus bridges the gap between the discretization accessible by `ddc::DiscreteElement` and the actual physical value of the coordinate.
+
+### ddc::DiscreteVector
+
+A `ddc::DiscreteVector` corresponds to an integer that, like `ddc::DiscreteElement`, carries the strong typing of the dimension in which it is defined. For instance in the uniform heat equation example, defining the `ddc::DiscreteVector` `gwx` as follows:
+
+```cpp
+ddc::DiscreteVector<DDimX> gwx(1);
+```
+
+is equivalent to defining a number of points, here 1, along the `x` dimension.
+
+> Note that the difference between two `ddc::DiscreteElement` creates a `ddc::DiscreteVector`, and the sum of a `ddc::DiscreteVector` and a `ddc::DiscreteElement` results in a `ddc::DiscreteElement`. This illustrates how `ddc::DiscreteElement` could correspond to points in an affine space, while `ddc::DiscreteVector` could correspond to vectors in a vector space or to a distance between two points.
```

Comment:
```suggestion
\remark Note that the difference between two `ddc::DiscreteElement` creates a `ddc::DiscreteVector`, and the sum of a `ddc::DiscreteVector` and a `ddc::DiscreteElement` results in a `ddc::DiscreteElement`. This illustrates how `ddc::DiscreteElement` could correspond to points in an affine space, while `ddc::DiscreteVector` could correspond to vectors in a vector space or to a distance between two points.
```

I think it shall be better illustrated with code than sentences.

Reply:
Ok